### PR TITLE
feat: digest view

### PR DESCRIPTION
**Pull request by tuvshinot**
Front end task Which is called Digest view( Modal like container shows up from left side)

### CODE REVIEW

File: js_machine_front/src/components/DigestView/components/DateAndView/DateAndView.tsx [hunk 1]
```diff
@@ -0,0 +1,18 @@
+import React, { memo } from 'react';
+import Eye from 'images/DigestView/eye.svg';
+import './DateAndView.css';
+
+interface Props {
+  views: number;
+}
+
+const DateAndView: React.FC<Props> = memo((props: Props) => {
```
Comment (vharadkou) [inline js_machine_front/src/components/DigestView/components/DateAndView/DateAndView.tsx]: please don't use default export,
export const DateAndView: ...

please check other places

File: js_machine_front/src/routes/News.tsx [hunk 2]
```diff
@@ -1,16 +1,48 @@
...
+
+  openDigest = () => {
+    this.setState({ openDigest: true });
+  }
+
+  closeDigest = () => {
+    this.setState({ openDigest: false });
+  }
+
+  exitKeyPressHandler = (e: any) => {
```
Comment (vharadkou) [inline js_machine_front/src/routes/News.tsx]: don't use any, try to find correct type
the same for other places in code

**Review by vharadkou — CHANGES_REQUESTED** [review]

**Comment by vharadkou** [comment]
btw, you have some conflicts

File: js_machine_front/src/components/DigestView/components/DateAndView/DateAndView.tsx (hunk 1 above)
Comment (tuvshinot) [reply js_machine_front/src/components/DigestView/components/DateAndView/DateAndView.tsx]: okey

File: js_machine_front/src/routes/News.tsx (hunk 2 above)
Comment (tuvshinot) [reply js_machine_front/src/routes/News.tsx]: okey

Comment (tuvshinot) [reply js_machine_front/src/routes/News.tsx]: hey, what kind of event is that? i thought it was KeyboardEvent but it gives me some error that says KeyboardEvent is missing keycode etc all properties )

File: js_machine_front/src/components/DigestView/DigestView.css [hunk 3]
```diff
@@ -0,0 +1,154 @@
+.DigestView {
```
Comment (vharadkou) [inline js_machine_front/src/components/DigestView/DigestView.css]: we are using BEM naming for now, which have the following convention (block-name_element-name)
you can take a look how to name it in other component

File: js_machine_front/src/components/DigestView/DigestView.tsx [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+        exitActive: 'DigestView-closed',
+      }}>
+      <div className="DigestView">
+        <div className="left-side">
+          <LeftArrowButton closeDigest={props.closeDigest} text="назад" />
+        </div>
+        <Option someMethod={() => alert('Btn Handlers')} />
+        <div className="right-side">
+          <DateAndView views={25} />
+          <h1>Digest Cycle #12</h1>
```
Comment (vharadkou) [inline js_machine_front/src/components/DigestView/DigestView.tsx]: please don't hardcode values, better to pass them thought props

File: js_machine_front/src/components/DigestView/DigestView.tsx [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+        enter: '',
+        enterActive: 'DigestView-open',
+        exit: '',
+        exitActive: 'DigestView-closed',
+      }}>
+      <div className="DigestView">
+        <div className="left-side">
+          <LeftArrowButton closeDigest={props.closeDigest} text="назад" />
+        </div>
+        <Option someMethod={() => alert('Btn Handlers')} />
```
Comment (vharadkou) [inline js_machine_front/src/components/DigestView/DigestView.tsx]: please don't use alerts

File: js_machine_front/src/components/DigestView/DigestView.tsx [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+        <div className="right-side">
+          <DateAndView views={25} />
```
Comment (vharadkou) [inline js_machine_front/src/components/DigestView/DigestView.tsx]: 25 is a magic number, create a variable with a corresponding name

File: js_machine_front/src/scenes/news/news.tsx [hunk 7]
```diff
@@ -1,18 +1,48 @@
...
+  paddingTop: 100,
+};
+
+interface State {
+  openDigest: boolean;
+}
+
+export class News extends PureComponent<{}, State> {
+
+  constructor(props: {}) {
```
Comment (vharadkou) [inline js_machine_front/src/scenes/news/news.tsx]: please remove constructor, you can define you state with initial values as a public property

File: js_machine_front/src/scenes/news/news.tsx [hunk 8]
```diff
@@ -1,18 +1,48 @@
...
+  openDigest = () => {
+    this.setState({ openDigest: true });
+  }
+
+  closeDigest = () => {
+    this.setState({ openDigest: false });
+  }
+
+  exitKeyPressHandler = (e: KeyboardEvent) => {
+    if (e.keyCode === 27) {
```
Comment (vharadkou) [inline js_machine_front/src/scenes/news/news.tsx]: please write a comment for which button this keyCode is responsible for

**Review by vharadkou — CHANGES_REQUESTED** [review]

**Comment by vharadkou** [comment]
[Digest view](https://trello.com/c/tNIYdnhM/32-digest-view)

File: js_machine_front/src/scenes/news/news.tsx [hunk 9]
```diff
@@ -1,18 +1,48 @@
...
+      this.closeDigest();
+    }
+  }
+
+  render(): JSX.Element {
+    return (
+      <div style={sectionStyle}>
+        <h1>News</h1>
+        <button onClick={this.openDigest}>Open digest</button>
+        <DigestView isOpen={this.state.openDigest}
```
Comment (vharadkou) [inline js_machine_front/src/scenes/news/news.tsx]: please create a new scene for digest view!!!

**Review by vharadkou — CHANGES_REQUESTED** [review]

File: js_machine_front/src/scenes/news/news.tsx (hunk 9 above)
Comment (vharadkou) [reply js_machine_front/src/scenes/news/news.tsx]: this is not the same as news, you need to create a separate route

Comment (tuvshinot) [reply js_machine_front/src/scenes/news/news.tsx]: okey

File: js_machine_front/src/scenes/digestView/styles/leftArrowButton.css [hunk 10]
```diff
@@ -1,17 +1,17 @@
-.LeftArrowButton {
+.digestview__leftArrowButton {
     position: fixed;
     top: 7%;
     left: 3%;
     cursor: pointer;
 }
 
-.LeftArrowButton-text {
+.digestview__leftArrowButton-text {
```
Comment (vharadkou) [inline js_machine_front/src/scenes/digestView/styles/leftArrowButton.css]: if you define styles for element inside another element you should write like left-arrow-button__text,
block-element-element is forbidden

File: js_machine_front/src/scenes/digestView/styles/dateAndView.css [hunk 11]
```diff
@@ -1,15 +1,15 @@
-.DateAndView {
+.digestview__dateAndView {
     display: flex;
     align-items: center;
 }
 
-.DateAndView-Img {
+.digestview__dateAndView-img {
```
Comment (vharadkou) [inline js_machine_front/src/scenes/digestView/styles/dateAndView.css]: btw, rename digestview to digest-view

**Review by vharadkou — CHANGES_REQUESTED** [review]

File: js_machine_front/src/scenes/digestView/components/dateAndView.tsx [hunk 12]
```diff
@@ -5,9 +5,9 @@ import { DateAndViewProps } from '../models/dateAndView';
 
 export const DateAndView: React.FC<DateAndViewProps> = memo((props: DateAndViewProps) => {
   return (
-    <div className="digestview__dateAndView">
-      <span className="digestview__dateAndView-date">5 АВГ</span>
-      <span><img src={Eye} alt="Eye" className="digestview__dateAndView-img" /> {props.views}</span>
+    <div className="dateAndView">
```
Comment (vharadkou) [inline js_machine_front/src/scenes/digestView/components/dateAndView.tsx]: date-and-view

**Review by vharadkou — CHANGES_REQUESTED** [review]

**Review by vharadkou — APPROVED** [review]

**Pull request merged**